app/src/main/java/org/wikipedia/descriptions/DescriptionEditView.kt

```diff
@@ -0,0 +1,371 @@
+        if (text != null && originalDescription != null) {
+            val scrollDelayMs = 500
+            postDelayed({ highlightEditText(binding.viewDescriptionEditText, originalDescription!!, text) }, scrollDelayMs.toLong())
+        }
+    }
+
+    private fun updateSaveButtonEnabled() {
+        if (!binding.viewDescriptionEditText.text.isNullOrEmpty() &&
+                originalDescription.orEmpty() != binding.viewDescriptionEditText.text.toString() &&
+                isTextValid) {
+            enableSaveButton(enabled = true, saveInProgress = false)
+        } else {
+            enableSaveButton(enabled = false, saveInProgress = false)
+        }
+    }
+
+    private fun enableSaveButton(enabled: Boolean, saveInProgress: Boolean) {
+        if (saveInProgress) {
+            binding.viewDescriptionEditSaveButton.setImageResource(R.drawable.ic_check_circle_black_24dp)
+            ImageViewCompat.setImageTintList(binding.viewDescriptionEditSaveButton, ColorStateList.valueOf(getThemedColor(context, R.attr.themed_icon_color)))
+            binding.viewDescriptionEditSaveButton.isEnabled = false
+            binding.viewDescriptionEditSaveButton.alpha = 1 / 2f
+        } else {
+            binding.viewDescriptionEditSaveButton.alpha = 1f
+            if (enabled) {
+                binding.viewDescriptionEditSaveButton.setImageResource(R.drawable.ic_check_circle_black_24dp)
+                ImageViewCompat.setImageTintList(binding.viewDescriptionEditSaveButton, ColorStateList.valueOf(getThemedColor(context, R.attr.themed_icon_color)))
+                binding.viewDescriptionEditSaveButton.isEnabled = true
+            } else {
+                binding.viewDescriptionEditSaveButton.setImageResource(R.drawable.ic_check_black_24dp)
+                ImageViewCompat.setImageTintList(binding.viewDescriptionEditSaveButton, ColorStateList.valueOf(getThemedColor(context, R.attr.material_theme_de_emphasised_color)))
+                binding.viewDescriptionEditSaveButton.isEnabled = false
+            }
+        }
+    }
+
+    fun showProgressBar(show: Boolean) {
+        binding.viewDescriptionEditProgressBar.visibility = if (show) VISIBLE else GONE
+    }
+
+    fun setAction(action: DescriptionEditActivity.Action) {
+        this.action = action
+        isTranslationEdit = action == DescriptionEditActivity.Action.TRANSLATE_CAPTION || action == DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION
+    }
+
+    private fun performReadArticleClick() {
+        callback?.onBottomBarClick()
+    }
+
+    override fun onLanguageDetectionSuccess(languageCode: String) {
+        if (languageCode != pageSummaryForEdit.lang &&
+                languageCode != WikipediaApp.getInstance().language().getDefaultLanguageCode(pageSummaryForEdit.lang)) {
+            isLanguageWrong = true
+            enqueueValidateText()
+        }
+    }
+
+    companion object {
+        private const val TEXT_VALIDATE_DELAY_MILLIS = 1000
```

Comment:
Would it be better if change to `private const val TEXT_VALIDATE_DELAY_MILLIS = 1000L` and reduce `toLong()`